Stream_support/include/CGAL/IO/io.h

```diff
@@ -267,6 +267,19 @@ class Input_rep
   std::istream& operator()( std::istream& is) const { return (is >> t); }
 };
 
+template <class T>
+class Input_rep<std::optional<T>>
+{
+  std::optional<T>& t;
+
+public:
+  //! initialize with a reference to \a t.
+  Input_rep( std::optional<T>& tt) : t(tt) {}
+
+  //! perform the input, calls \c operator\>\> by default.
+  std::istream& operator()( std::istream& is) const { return (is >> t.value()); }
+};
```

Comment:
According to the [documentation of `optional::value()`](https://en.cppreference.com/w/cpp/utility/optional/value):
> If `*this` contains a value, returns a reference to the contained value.
>  Otherwise, throws a [`std::bad_optional_access`].(https://en.cppreference.com/w/cpp/utility/optional/bad_optional_access) exception.

If the `optional` is empty, then `is >> t.value()` will throw.

I suggest:
```suggestion
template <class T>
class Input_rep<std::optional<T>>
{
  std::optional<T>& t;

public:
  //! initialize with a reference to \a t.
  Input_rep( std::optional<T>& tt) : t(tt) {}

  //! perform the input, calls \c operator\>\> by default.
  std::istream& operator()( std::istream& is) const {
    T v;
    if(is >> v) t = v;
    return is;
  }
};
```